Graphics/Vulkan/VulkanGraphicsPluginProvider.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#include <NovelRT/Graphics/GraphicsAdapter.hpp>
+#include <NovelRT/Graphics/GraphicsSurfaceContext.hpp>
+#include <NovelRT/Graphics/IGraphicsSurface.hpp>
+#include <NovelRT/Graphics/Vulkan/VulkanGraphicsAdapterSelector.hpp>
+#include <NovelRT/Graphics/Vulkan/VulkanGraphicsBackendTraits.hpp>
+#include <NovelRT/Graphics/Vulkan/VulkanGraphicsProvider.hpp>
+#include <NovelRT/PluginManagement/IGraphicsPluginProvider.h>
+
+// this is such a hack
+using namespace NovelRT::PluginManagement;
+using namespace NovelRT::Graphics::Vulkan;
+using namespace NovelRT::Graphics;
+
+template<>
+IGraphicsPluginProvider<VulkanGraphicsBackend>::IGraphicsPluginProvider() noexcept
+    : _graphicsProvider([]() { return std::make_shared<VulkanGraphicsProvider>(); })
+{
+}
+
+template<>
+[[nodiscard]] std::shared_ptr<GraphicsProvider<VulkanGraphicsBackend>> IGraphicsPluginProvider<
+    VulkanGraphicsBackend>::GetGraphicsProvider()
+{
+    //TODO: unfuck this
+    return std::make_shared<GraphicsProvider<VulkanGraphicsBackend>>(_graphicsProvider.getActual());
+}
+
+template<>
+std::shared_ptr<GraphicsSurfaceContext<VulkanGraphicsBackend>> IGraphicsPluginProvider<
+    VulkanGraphicsBackend>::CreateSurfaceContext(std::shared_ptr<IGraphicsSurface> surface)
+{
+    //TODO: unfuck this
+    return std::make_shared<GraphicsSurfaceContext<VulkanGraphicsBackend>>(nullptr, surface, GetGraphicsProvider());
+}
```

Comment:
Could we maybe... unfuck these now? Or are they just cursed but guaranteed to work? If that's the case, I'm okay with deferring, just the comments seem... nefarious.

Reply:
This is because of the changes we made to how the graphics plugin was loaded. Comments are still relevant but code itself is perfectly safe and works as intended for the time being.